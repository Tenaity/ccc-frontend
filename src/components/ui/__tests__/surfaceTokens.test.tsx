import "@testing-library/jest-dom/vitest"

import { render, screen } from "@testing-library/react"
import { beforeAll, describe, expect, test, vi } from "vitest"

import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "../select"
import { Popover, PopoverContent, PopoverTrigger } from "../popover"
import { Button } from "../button"

beforeAll(() => {
  const protoTargets: Array<Record<string, unknown>> = []

  if (typeof window.Element !== "undefined") {
    protoTargets.push(window.Element.prototype as unknown as Record<string, unknown>)
  }

  if (typeof window.HTMLElement !== "undefined") {
    protoTargets.push(
      window.HTMLElement.prototype as unknown as Record<string, unknown>,
    )
  }

  for (const proto of protoTargets) {
    Object.defineProperty(proto, "scrollIntoView", {
      configurable: true,
      writable: true,
      value: vi.fn(),
    })
  }
})

describe("surface tokens", () => {
  test("SelectContent uses popover surface token", async () => {
    render(
      <Select
        value="one"
        onValueChange={() => undefined}
        open
        onOpenChange={() => undefined}
      >
        <SelectTrigger aria-label="Chọn mục">
          <SelectValue placeholder="Chọn" />
        </SelectTrigger>
        <SelectContent>
          <SelectItem value="one">Một</SelectItem>
          <SelectItem value="two">Hai</SelectItem>
        </SelectContent>
      </Select>,
    )

    const listbox = await screen.findByRole("listbox")
    expect(listbox).toHaveClass("bg-popover")
    expect(listbox).toMatchSnapshot()
  })

  test("PopoverContent uses popover surface token", async () => {
    render(
      <Popover open>
        <PopoverTrigger asChild>
          <Button type="button">Mở</Button>
        </PopoverTrigger>
        <PopoverContent data-testid="popover-content">Nội dung</PopoverContent>
      </Popover>,
    )

    const content = await screen.findByTestId("popover-content")
    expect(content).toHaveClass("bg-popover")
    expect(content).toMatchSnapshot()
  })
})
