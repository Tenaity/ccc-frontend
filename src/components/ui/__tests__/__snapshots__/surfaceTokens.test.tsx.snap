// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`surface tokens > PopoverContent uses popover surface token 1`] = `
<div
  class="z-50 w-72 rounded-md border border-border bg-popover p-4 text-popover-foreground shadow-md outline-none data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 origin-[--radix-popover-content-transform-origin]"
  data-align="center"
  data-side="bottom"
  data-state="open"
  data-testid="popover-content"
  id="radix-:r3:"
  role="dialog"
  style="--radix-popover-content-transform-origin: var(--radix-popper-transform-origin); --radix-popover-content-available-width: var(--radix-popper-available-width); --radix-popover-content-available-height: var(--radix-popper-available-height); --radix-popover-trigger-width: var(--radix-popper-anchor-width); --radix-popover-trigger-height: var(--radix-popper-anchor-height);"
  tabindex="-1"
>
  Nội dung
</div>
`;

exports[`surface tokens > SelectContent uses popover surface token 1`] = `
<div
  class="relative z-50 max-h-[--radix-select-content-available-height] min-w-[8rem] overflow-y-auto overflow-x-hidden rounded-md border border-border bg-popover text-popover-foreground shadow-md data-[state=open]:animate-in data-[state=closed]:animate-out data-[state=closed]:fade-out-0 data-[state=open]:fade-in-0 data-[state=closed]:zoom-out-95 data-[state=open]:zoom-in-95 data-[side=bottom]:slide-in-from-top-2 data-[side=left]:slide-in-from-right-2 data-[side=right]:slide-in-from-left-2 data-[side=top]:slide-in-from-bottom-2 origin-[--radix-select-content-transform-origin] data-[side=bottom]:translate-y-1 data-[side=left]:-translate-x-1 data-[side=right]:translate-x-1 data-[side=top]:-translate-y-1"
  data-align="start"
  data-side="bottom"
  data-state="open"
  dir="ltr"
  id="radix-:r0:"
  role="listbox"
  style="box-sizing: border-box; display: flex; flex-direction: column; outline: none; --radix-select-content-transform-origin: var(--radix-popper-transform-origin); --radix-select-content-available-width: var(--radix-popper-available-width); --radix-select-content-available-height: var(--radix-popper-available-height); --radix-select-trigger-width: var(--radix-popper-anchor-width); --radix-select-trigger-height: var(--radix-popper-anchor-height); pointer-events: auto;"
  tabindex="-1"
>
  <style>
    [data-radix-select-viewport]{scrollbar-width:none;-ms-overflow-style:none;-webkit-overflow-scrolling:touch;}[data-radix-select-viewport]::-webkit-scrollbar{display:none}
  </style>
  <div
    class="p-1 h-[var(--radix-select-trigger-height)] w-full min-w-[var(--radix-select-trigger-width)]"
    data-radix-select-viewport=""
    role="presentation"
    style="position: relative; flex: 1 1 0%; overflow: hidden auto;"
  >
    <div
      aria-labelledby="radix-:r1:"
      aria-selected="true"
      class="relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-2 pr-8 text-sm outline-none focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
      data-highlighted=""
      data-radix-collection-item=""
      data-state="checked"
      role="option"
      tabindex="-1"
    >
      <span
        class="absolute right-2 flex h-3.5 w-3.5 items-center justify-center"
      >
        <span
          aria-hidden="true"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-check h-4 w-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M20 6 9 17l-5-5"
            />
          </svg>
        </span>
      </span>
      <span
        id="radix-:r1:"
      >
        Một
      </span>
    </div>
    <div
      aria-labelledby="radix-:r2:"
      aria-selected="false"
      class="relative flex w-full cursor-default select-none items-center rounded-sm py-1.5 pl-2 pr-8 text-sm outline-none focus:bg-accent focus:text-accent-foreground data-[disabled]:pointer-events-none data-[disabled]:opacity-50"
      data-radix-collection-item=""
      data-state="unchecked"
      role="option"
      tabindex="-1"
    >
      <span
        class="absolute right-2 flex h-3.5 w-3.5 items-center justify-center"
      />
      <span
        id="radix-:r2:"
      >
        Hai
      </span>
    </div>
  </div>
</div>
`;
